File: Models/SQ_Soil_Temperature/src/java/SQ_Soil_Temperature/SoilTemperatureRate.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class SoilTemperatureRate
{
    private Double heatFlux;
    
    public SoilTemperatureRate() { }
    
    public SoilTemperatureRate(SoilTemperatureRate toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.heatFlux = toCopy.getheatFlux();
        }
    }
    public Double getheatFlux()
    { return heatFlux; }

    public void setheatFlux(Double _heatFlux)
    { this.heatFlux= _heatFlux; } 
    
}

File: Models/SQ_Soil_Temperature/src/cpp/SQ_Soil_Temperature/SoilTemperatureRate.cpp
#include "SoilTemperatureRate.h"
using namespace SQ_Soil_Temperature;

SoilTemperatureRate::SoilTemperatureRate() {}

double SoilTemperatureRate::getheatFlux() { return this->heatFlux; }

void SoilTemperatureRate::setheatFlux(double _heatFlux) { this->heatFlux = _heatFlux; }